src/v/cluster/rm_stm.cc

```diff
@@ -2750,68 +2750,69 @@ ss::future<stm_snapshot> rm_stm::take_snapshot() {
         });
     }
 
-    iobuf tx_ss_buf;
-    auto version = active_snapshot_version();
-    if (version == tx_snapshot::version) {
-        tx_snapshot tx_ss;
-        fill_snapshot_wo_seqs(tx_ss);
-        for (const auto& entry : _log_state.seq_table) {
-            tx_ss.seqs.push_back(entry.second.entry.copy());
-        }
-        tx_ss.offset = _insync_offset;
-
-        for (const auto& entry : _log_state.tx_seqs) {
-            tx_ss.tx_seqs.push_back(tx_snapshot::tx_seqs_snapshot{
-              .pid = entry.first, .tx_seq = entry.second});
-        }
-
-        for (const auto& entry : _log_state.expiration) {
-            tx_ss.expiration.push_back(tx_snapshot::expiration_snapshot{
-              .pid = entry.first, .timeout = entry.second.timeout});
-        }
-
-        reflection::adl<tx_snapshot>{}.to(tx_ss_buf, std::move(tx_ss));
-    } else if (version == tx_snapshot_v2::version) {
-        tx_snapshot_v2 tx_ss;
-        fill_snapshot_wo_seqs(tx_ss);
-        for (const auto& entry : _log_state.seq_table) {
-            tx_ss.seqs.push_back(entry.second.entry.copy());
-        }
-        tx_ss.offset = _insync_offset;
-        reflection::adl<tx_snapshot_v2>{}.to(tx_ss_buf, std::move(tx_ss));
-    } else if (version == tx_snapshot_v1::version) {
-        tx_snapshot_v1 tx_ss;
-        fill_snapshot_wo_seqs(tx_ss);
-        for (const auto& it : _log_state.seq_table) {
-            auto& entry = it.second.entry;
-            seq_entry_v1 seqs;
-            seqs.pid = entry.pid;
-            seqs.seq = entry.seq;
-            try {
-                seqs.last_offset = to_log_offset(entry.last_offset);
-            } catch (...) {
-                // ignoring outside the translation range errors
-                continue;
+    return f.then([this]() mutable {
```

Comment:
@ZeDRoman 

> tx: fix race in take_snapshot
offload_aborted_txns changes _log_state.aborted
fill_snapshot_wo_seqs uses _log_state.aborted
they shouldn't work in parallel

please write detailed commit messages that describe how you are fixing an issue. all this commit message says is that two things are no longer updated in parallel. it's not obvious what is happening out of the >100 LOC.

Reply:
the commit contains only change place of f continuation
other lines is just a clang-format

Reply:
>the commit contains only change place of f continuation
other lines is just a clang-format

1. when the changes are this big due to clang-format, then split the commit so the formatting-only change is clear
2. the commit message still should have said what changed to remove the unwanted parallelism.